### PR TITLE
more e2e tests for secret scrubbing

### PR DESCRIPTION
**Pull request by taylordowns2000**
## Notes for the reviewer

@jyeshe , i've extended the e2e tests to ensure that secrets are scrubbed in downstream jobs (✅ ), that the safelist is still respected (✅ ) and that the base64 representations of those strings are still scrubbed... ❌ .

I'm hoping that this addition to the e2e test helps clarify the behavior of the scrubber! Please merge into your branch and fix the implementation (if needed) on that side.

Related to #1519

### CODE REVIEW

**Review by jyeshe — DISMISSED** [review]
:clap:

File: lib/lightning/scrubber.ex [hunk 1]
```diff
@@ -88,15 +96,18 @@ defmodule Lightning.Scrubber do
...
       samples
       |> Enum.filter(fn x -> not is_boolean(x) end)
       |> Enum.map(fn x -> if is_integer(x), do: Integer.to_string(x), else: x end)
 
     base64_secrets =
       stringified_samples
+      |> Enum.concat(usernames)
       |> cartesian_pairs()
       |> Enum.map(fn [x, y] ->
         "#{x}:#{y}"
```
Comment (jyeshe) [inline lib/lightning/scrubber.ex:113]: Username or email needed for matching Base64 auth method (Basic authentication) as tested here:
https://github.com/OpenFn/platform-app/blob/4d4b1d35d8abc491757ee192d8c22b26ebb80573/test/services/credential_service_test.exs#L73

**Review by jyeshe — APPROVED** [review]
round :two:

**Comment by taylordowns2000** [comment]
this is great @jyeshe - lovely catch on the basic auth composition

**Pull request merged**